siliconcompiler/remote/client.py

```diff
@@ -274,7 +275,26 @@ def remote_run(chip):
 
 ###################################
 def remote_run_loop(chip):
+    # Wrapper to allow for capturing of Ctrl+C
+    try:
+        __remote_run_loop(chip)
+    except KeyboardInterrupt:
+        jobid = chip.status['jobhash']
+        entry_step, entry_index = \
+            chip._get_flowgraph_entry_nodes(flow=chip.get('option', 'flow'))[0]
+        entry_manifest = os.path.join(chip._getworkdir(step=entry_step, index=entry_index),
+                                      'outputs',
+                                      f'{chip.design}.pkg.json')
+        reconnect_cmd = f'sc-remote -jobid {jobid} -cfg {entry_manifest} -reconnect'
+        cancel_cmd = f'sc-remote -jobid {jobid} -cancel'
+        chip.logger.info('Disconnecting from remote job')
+        chip.logger.info(f'To reconnect to this job use: {reconnect_cmd}')
+        chip.logger.info(f'To cancel this job use: {cancel_cmd}')
+        sys.exit(0)
```

Comment:
One reason we usually raise `SiliconCompilerError` instead of calling `sys.exit(1)` is that `sys.exit` can't be caught with blanket `except:` blocks. So, if somebody calls `chip.run()` in a script which has post-run logic, this exit will prevent most of that logic from running unless they specifically catch the `except SystemExit:` case.

That's probably fine, but how would you feel about `raise SiliconCompilerError("Job canceled by user keyboard interrupt")`?

Reply:
it's a good point. The stacktrace is kind of messy but I can try it out and see what it ends up looking like